src/components/SelectorSection/SortedSelector.tsx

```diff
@@ -1,25 +1,33 @@
-import React, { ChangeEvent } from 'react';
-import Store from '../../store';
+import React, { ChangeEvent, useCallback } from 'react';
+import { useLunchDispatch } from '../../hooks';
+import { CHANGE_SORT_STATE } from '../../store/action';
+import { Sort } from '../../store/type';
+
 import styles from './SelectorSection.module.css';
-import type { Sort, State } from '../../App';
 
-class SortedSelector extends React.PureComponent {
-	private static handleSortedSelector = (store: State | null) => (e: ChangeEvent<HTMLSelectElement>) => {
-		store?.setSortState(e.target.value as Sort);
-	};
+function SortedSelector() {
+  const dispatch = useLunchDispatch();
+
+  const handleSortedSelector = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      dispatch({
+        type: CHANGE_SORT_STATE,
+        payload: { sort: e.target.value as Sort },
+      });
+    },
+    []
+  );
```

Comment:
dispatch를 사용하는 좋은 방법 중 하나는, `useLunchDispatch`를 한 번 더 추상화하는 겁니다.
현재는 dispatch를 그대로 꺼내 사용하기 때문에 `CHANGE_SORT_STATE` 상수를 import 해야하고, 객체에 대한 key-value를 직접 정의해야하는 보일러 플레이트가 생기기 때문이에요. 아래와 같이 사용하면 어떨까요?

```suggestion
  const { setSortedType } = useLunch();

  const handleSortedSelector = useCallback(
    (e: ChangeEvent<HTMLSelectElement>) => {
      setSortedType(e.target.value as Sort);
    },
    []
  );
```

좀 더 관심사가 분리된 것 같지 않나요😁